test/testUtils.ts

```diff
@@ -30,16 +30,26 @@ export async function initTest() {
   if (!process.env.BSC_BUNDLER_URL) throw new Error('BSC_BUNDLER_URL is missing');
   if (!process.env.BSC_NODE_URL) throw new Error('BSC_NODE_URL is missing');
   if (!process.env.BSC_CHAIN_ID) throw new Error('BSC_CHAIN_ID is missing');
+  if (!process.env.POL_BUNDLER_URL) throw new Error('POL_BUNDLER_URL is missing');
+  if (!process.env.POL_NODE_URL) throw new Error('POL_NODE_URL is missing');
+  if (!process.env.POL_CHAIN_ID) throw new Error('POL_CHAIN_ID is missing');
   if (!process.env.MORALIS_API_KEY) throw new Error('MORALIS_API_KEY is missing');
 
   const chainConfigs: ChainConfigs = {
     [Number(process.env.ETH_CHAIN_ID)]: {
       rpcUrl: process.env.ETH_NODE_URL,
       bundlerUrl: process.env.ETH_BUNDLER_URL,
+      factory: '0x61e218301932a2550AE8E4Cd1EcfCA7bE64E57DC',
     },
     [Number(process.env.BSC_CHAIN_ID)]: {
       rpcUrl: process.env.BSC_NODE_URL,
       bundlerUrl: process.env.BSC_BUNDLER_URL,
+      factory: '0x61e218301932a2550AE8E4Cd1EcfCA7bE64E57DC',
+    },
+    [Number(process.env.POL_CHAIN_ID)]: {
+      rpcUrl: process.env.POL_NODE_URL,
+      bundlerUrl: process.env.POL_BUNDLER_URL,
+      factory: '0xd9a6d24030c0DFB0bf78170556a8B671Ec432AAC',
```

Comment:
we have factory addresses on src/constants

Reply:
factory is optional param, https://github.com/blndgs/intents.js/pull/111/files#diff-b11023b140fe883d52890186c3fe2e471bd8582bc73decf55a309f1b7056eca4L30 it chainconfig contains it then factory will be as per chainconfig else using the setting if from the FACTORY constant.

this one is for the test (custom chain ids)

Reply:
marking it resolve as discussed